packages/frontend-2/lib/viewer/composables/filtering/counts.ts

```diff
@@ -0,0 +1,189 @@
+import { FilteringExtension, type PropertyInfo, ViewerEvent } from '@speckle/viewer'
+import { useInjectedViewerState } from '~~/lib/viewer/composables/setup'
+import {
+  ExistenceFilterCondition,
+  type FilterData
+} from '~/lib/viewer/helpers/filters/types'
+
+/**
+ * Get count of filtered objects directly from the viewer
+ * Uses viewer events to stay in sync with the viewer's internal state
+ */
+export function useFilteredObjectsCount() {
+  const {
+    viewer,
+    ui: { filters }
+  } = useInjectedViewerState()
+  const filteredObjectsCount = ref(0)
```

Comment:
feels like this should be in the state. currently if you invoke this composable in multiple components, each are gonna have their own duplicate version of this